app/models/comment.rb

```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post, counter_cache: :comments_counter
+
+  def update_post_counter
+    post.update(comments_counter: post.comments.count)
+  end
+end
```

Comment:
OPTIONAL
- You can also consider using the helper method `after_save` to ensure that with every update related to this specific model `Comment` then invoke the `update_post_counter` [Please consider renaming this method to update_comment_counter for more clarity and readability] method immediately. 👍

Reply:
hi @MahmoudBakr23 thank you for your suggestions actually, I was adding it for requirement purposes, I was actually caching the counter in every update so the method is unnecessary